CBinding/Gui/CTextEditorExtension.cs

```diff
@@ -283,7 +284,7 @@ bool ShouldCompleteOn (char pressed)
 
 		public override Task<ICompletionDataList> HandleCodeCompletionAsync (CodeCompletionContext completionContext, CompletionTriggerInfo triggerInfo, CancellationToken token = default (CancellationToken))
 		{
-			var project = (CProject)DocumentContext.Project;
+			var project = (CMakeProject)(DocumentContext.Project as SolutionItem);
```

Comment:
I think Owner property in DocumentContext needs to be chained to Project . Some changes here :- 
https://github.com/mono/monodevelop/commit/f854095505fadeeeb97d8541d00d8a11f718395d